astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,84 @@
+from os import environ
+from pathlib import Path
+from tarfile import open as tar_open
+from tempfile import mkdtemp
+from warnings import warn
+
+from astropy.utils.data import download_file
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=mkdtemp()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Always download our data files fresh. Ensures changes from redcat team will be picked up
+    # Download HST Instrument Data Files
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz')) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        tar.extractall(path=SIM_DATA_DIR)
+    # Download Vega CALSPEC Reference Atlas
+    calspec_dir = SIM_DATA_DIR / 'grp' / 'redcat' / 'trds' / 'calspec'
+    calspec_dir.mkdir(parents=True, exist_ok=True)
+    vega_reference_atlas = Path(download_file('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/alpha_lyr_stis_010.fits')) # noqa
+    vega_reference_atlas.replace(calspec_dir / 'alpha_lyr_stis_010.fits')
+
+    # Now that we have all our reference files, we can import stsynphot
+    # (This is why it's not a top-line import)
+    from stsynphot import Vega, band # noqa
+    if grism == 'G141':
+        if detector:
+            warn("WFC3's G141 grism does not have multiple detectors. Ignoring detector argument",
+                 RuntimeWarning)
+        bandpass = band('wfc3,ir,g141')
+    elif grism == 'G102':
+        if detector:
+            warn("WFC3's G102 grism does not have multiple detectors. Ignoring detector argument",
+                 RuntimeWarning)
+        bandpass = band('wfc3,ir,g102')
+    elif grism == 'G280':
+        bandpass = band(f'wfc3,uvis{detector},g280')
+    elif grism == 'G800L':
+        bandpass = band(f'acs,wfc{detector},g800l')
+    else:
+        raise ValueError(f"Unrecognized grism: {grism}. Valid grisms: G141, G102, G280, G800L")
+
+    spectrum = (bandpass * Vega).to_spectrum1d()
```

Comment:
What is your intent here? Vega spectrum is not normalized. Usually people normalize it first. But if you are trying to predict Vega's count rate through the given bandpass here, you should use `synphot.Observation`.

Reply:
I've switched the calculation to call `synphot.Observation` and immediately convert it to a spectrum1d. Let me know if this makes more sense? https://github.com/spacetelescope/astrogrism/pull/18/files#diff-a44ad3e831c16508917324e007e1c88a870570691f40e4410b6fd5782c37b6a1R71